pkg/daemon/daemon.go

```diff
@@ -356,70 +419,26 @@ func (d *daemon) DeletePeriodicUpdate() {
 			continue
 		}
 
-		netAttInfo, err := d.kubeClient.GetNetworkAttachmentDefinition(networkNamespace, networkName)
+		networkName, ibCniSpec, err := d.GetIbSriovNetwork(networkID)
 		if err != nil {
-			log.Warn().Msgf("failed to get networkName attachment %s with error: %v", networkName, err)
-			// skip failed networks
-			continue
-		}
-		log.Debug().Msgf("networkName attachment %v", netAttInfo)
-
-		networkSpec := make(map[string]interface{})
-		err = json.Unmarshal([]byte(netAttInfo.Spec.Config), &networkSpec)
-		if err != nil {
-			log.Warn().Msgf("failed to parse networkName attachment %s with error: %v", networkName, err)
-			// skip failed networks
-			continue
-		}
-		log.Debug().Msgf("networkName attachment spec %+v", networkSpec)
-
-		ibCniSpec, err := utils.GetIbSriovCniFromNetwork(networkSpec)
-		if err != nil {
-			log.Warn().Msgf("failed to get InfiniBand SR-IOV CNI spec from network attachment %+v, with error: %v",
-				networkSpec, err)
-			// skip failed networks
+			log.Err(err)
 			continue
 		}
-		log.Debug().Msgf("CNI spec %+v", ibCniSpec)
 
 		var guidList []net.HardwareAddr
 		var failedPods []*kapi.Pod
+		var guidAddr net.HardwareAddr
 		for _, pod := range pods {
 			log.Debug().Msgf("pod namespace %s name %s", pod.Namespace, pod.Name)
-			networks, netErr := netAttUtils.ParsePodNetworkAnnotation(pod)
-			if netErr != nil {
-				failedPods = append(failedPods, pod)
-				log.Error().Msgf("failed to read pod networkName annotations pod namespace %s name %s, with error: %v",
-					pod.Namespace, pod.Name, netErr)
-				continue
-			}
-
-			network, netErr := utils.GetPodNetwork(networks, networkName)
-			if netErr != nil {
-				failedPods = append(failedPods, pod)
-				log.Error().Msgf("failed to get pod networkName spec %s with error: %v", networkName, netErr)
-				// skip failed pod
-				continue
-			}
-
-			if !utils.IsPodNetworkConfiguredWithInfiniBand(network) {
-				log.Warn().Msgf("network %+v is not InfiniBand configured", network)
-				continue
-			}
-
-			allocatedGUID, netErr := utils.GetPodNetworkGUID(network)
-			if netErr != nil {
-				failedPods = append(failedPods, pod)
-				log.Err(netErr)
+			guidAddr, err = getPodGUIDForNetwork(pod, networkName)
+			if err != nil {
+				if !strings.Contains(err.Error(), "is not Infiniband configured") {
```

Comment:
I think we should have an error construct with error code in addition to error message.
i see `strings.Contains()` appear in several places

this is something that did not exist prior to this refactor.

Reply:
Like this?
```
type errorObj struct {
    code       int
    instance error
}
```

Reply:
you should create a new package `errors`

```
package errors

import "fmt"

const (
    UNKNOWN_ERROR = iota
    // add error codes here
)

type CodedError struct {
    message string
    code int
}

func (e *CodedError)Error() string {
    return e.message
}

func Errorf(code uint, msg, args...) error {
    return CodedError{message: fmt.Sprintf(msg, args), code: code}
}

func CodeFromError(err error) int {
    switch e := err.(type) {
    case CodedError:
        return e.code
    default:
        return UNKNOWN_ERROR
    }
}
```

i dont guarantee the above compiles or works :) but it should give you a general idea

Reply:
Ok. I'll put it in separate commit.